ext/cgi/escape/escape.c

```diff
@@ -0,0 +1,94 @@
+#include "ruby.h"
+#include "ruby/encoding.h"
+
+static VALUE rb_cCGI, rb_mUtil, rb_mEscape;
+
+static VALUE
+html_escaped_cat(VALUE str, char c)
+{
+    switch (c) {
+      case '\'':
+	rb_str_cat_cstr(str, "&#39;");
+	break;
+      case '&':
+	rb_str_cat_cstr(str, "&amp;");
+	break;
+      case '"':
+	rb_str_cat_cstr(str, "&quot;");
+	break;
+      case '<':
+	rb_str_cat_cstr(str, "&lt;");
+	break;
+      case '>':
+	rb_str_cat_cstr(str, "&gt;");
+	break;
+    }
+}
+
+static VALUE
+optimized_escape_html(VALUE str)
+{
+    long i, len, modified = 0, beg = 0, offset = 0;
```

Comment:
Fixed in https://github.com/k0kubun/ruby/commit/87a0be2a03977c72c6a96a4897b9477d79e27e33.